### DISCUSSION TITLE
Translations - Character Limit?

### DISCUSSION THREAD

**Discussion by ERYpTION** [streetcomplete/StreetComplete#4047] [Q&A]
I have been doing some proofreading and new translations. Sometimes I struggle to keep the translation as short as the original string. Mostly I think it's fine but on some screens I worry it will break the layout. For example on the screen with the different types of bicycle paths I wonder if there is a character limit for the text beneath each image or if the layout can handle more than two lines of text. E.g. the string "sidepath or protected bike lane" is only 31 characters but I would like to use about 50 for the translation. Will the layout be able to handle that?

**Comment by matkoniecz** [comment]
It depends on a phone. In general Polish translation has a bit longer text than English one.

It is worth checking - for example text on views with contrained layout (building level quest) will run into issues faster than message popup where doubling or trippling text length may be fine.

If you use tablet or massive phone you can borrow phone of friend/relative and test there once app is released. Or use emulator in Android Studio.

Or ask people in your OSM community to look for problems.

> Sometimes I struggle to keep the translation as short as the original string. 

This is not strictly needed.

-----------------

BTW, if text breaks interface AND there is no way to shorten it - then it may be worth to open an issue requesting to improve layout. But first contact some other speaker of language to help in rephrasing and shortening.

**Comment by ERYpTION** [comment]
Alright, I have added the (longer) translation and will try to monitor for negative feedback after the next release.
